rules/coding-style/src/Rector/Throw_/AnnotateThrowablesRector.php

```diff
@@ -93,59 +117,77 @@ public function throwException(int $code)
         );
     }
 
-    /**
-     * @param Throw_ $node
-     */
     public function refactor(Node $node): ?Node
     {
-        if ($this->isThrowableAnnotated($node)) {
-            return null;
+        $this->throwablesToAnnotate = [];
+        if ($this->hasThrowablesToAnnotate($node)) {
+            $this->annotateThrowables($node);
+            return $node;
         }
 
-        $this->annotateThrowable($node);
-
-        return $node;
+        return null;
     }
 
-    private function isThrowableAnnotated(Throw_ $throw): bool
+    private function hasThrowablesToAnnotate(Node $node): bool
     {
-        $phpDocInfo = $this->getThrowingStmtPhpDocInfo($throw);
-        $identifiedThrownThrowables = $this->identifyThrownThrowables($throw);
+        if ($node instanceof Throw_) {
+            $this->analyzeStmtThrow($node);
```

Comment:
This method should return number of found throws, instead of using property that is filled by this method.

Reply:
I wrote this this way because in `refactor()` method it is more readable.

It sounds like `if hasThrowablesToAnnotate then annotateThrowables`.

The method name is a question and implies a true/false answer: "does this node have throwables to annotate? Yes or no?".

If I return the number of found throwables to annotate, then I need to change the name to something like `foundThrowablesToAnnotate` and then, in `refactor()`, change the `if` to something like `if (0 < $this->foundThrowablesToAnnotate()`.

but I think this way of writing code is less discursive and I think that when the code is discursive is easier to be read.

But if you think this is really important I will refactor the methods.

Reply:
* How would you test this method if decoupled to service?

It's better to use input/output way of coding that shows the result. Then work with result explicitly in this flow of code. Now I have to go inside the method, to figure out what "analyse" really does.

Reply:
It might sound I'm making big deal of it. But other contributors might read your code and copy-paste the pattern to their rules, which would eventually lead to hard to maintain legacy code.

Reply:
Reading again I’ve just understood you are speaking about analyze*() methods and not abou HasThrowablesToAnnotate() 😓

I will change them tomorrow...